### ISSUE TITLE
CA3075 false positives for XmlReaderSettings

### ISSUE THREAD

**Issue by davidwengier** [dotnet/roslyn-analyzers#2105]
#### Analyzer package

[Microsoft.NetFramework.Analyzers](https://www.nuget.org/packages/Microsoft.NetFramework.Analyzers)

#### Package Version

v2.6.3

#### Diagnostic ID

[CA3075](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca3075-insecure-dtd-processing)

#### Repro steps

```
using System.IO;
using System.Xml;

namespace ConsoleApp8
{
    class Program
    {
        private static readonly XmlReaderSettings s_xmlReaderSettings = new XmlReaderSettings()
        {
            DtdProcessing = DtdProcessing.Prohibit,
            XmlResolver = null
        };

        static void Main(string[] args)
        {
            var settings = new XmlReaderSettings()
            {
                DtdProcessing = DtdProcessing.Prohibit,
                XmlResolver = null
            };
            using (var reader = new StringReader(@"<xml></xml>"))
            using (var xmlReader = XmlReader.Create(reader, settings))                   // no warning given
            {
            }


            using (var reader = new StringReader(@"<xml></xml>"))
            using (var xmlReader = XmlReader.Create(reader, s_xmlReaderSettings))        // false positive warning
            {
            }

            using (var reader = new StringReader(@"<xml></xml>"))
            using (var xmlReader = XmlReader.Create(reader, new XmlReaderSettings()      // false positive warning
            {
                DtdProcessing = DtdProcessing.Prohibit,
                XmlResolver = null
            }))
            {
            }
        }
        
        private readonly XmlReaderSettings _xmlReaderSettingsInstance = new XmlReaderSettings()
        {
            DtdProcessing = DtdProcessing.Prohibit,
            XmlResolver = null
        };

        void Instance()
        {
            using (var reader = new StringReader(@"<xml></xml>"))
            using (var xmlReader = XmlReader.Create(reader, _xmlReaderSettingsInstance)) // false positive warning
            {
            }
        }
    }
}
```

#### Expected behavior

No warnings.

#### Actual behavior

Inconsistent warnings on the final three examples in the above. Somewhat annoyingly, the last example gets a false positive, and it is the example used as the solution in the documentation.

**Comment by dotpaul** [comment]
This should hopefully be fixed as part of #1591, when rewriting this analyzer to take advantage of dataflow analysis.

**Comment by dotpaul** [comment]
In the meantime, I'll do a short term fix in the v2.6 branch that err on the side of false negatives instead of false positives.  As it is now, the analyzer assumes insecure XmlReaderSettings when it doesn't have any information from looking at the method body.
